### PR TITLE
feat: add ability to post identity key via admin API

### PR DESCRIPTION
**Pull request by austin-king**
## High Level Overview of Change

Asana task: https://app.asana.com/0/1182225689526777/1186929984905884

### Context of Change

Part of a series of PRs adding user supplied identity keys & signatures to addresses to unlock non custodial wallet partners

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

## Before / After

Extending the `insertUser` function to an accept an optional identityKey param

## Test Plan

I added a unit test confirming it works + doesn't break our old ones b/c it's an optional param

## Future Tasks

The subtasks outlined in this Asana task: https://app.asana.com/0/1182225689526777/1186312611097763

### CODE REVIEW

**Comment by austin-king** [comment]
Don't review yet, I broke my local spacing config and it's making my PRs ugly. Lemme fix before you waste your time

**Comment by austin-king** [comment]
> Don't review yet, I broke my local spacing config and it's making my PRs ugly. Lemme fix before you waste your time

@hbergren I knew you were somehow going to see this minutes after I pushed it, glad I left that comment lol

File: src/data-access/users.ts [hunk 1]
```diff
@@ -41,11 +42,13 @@ export async function checkUserExistence(payId: string): Promise<boolean> {
 export async function insertUser(
   payId: string,
   addresses: readonly AddressInformation[],
+  identityKey: string | undefined = undefined,
```
Comment (hbergren) [inline src/data-access/users.ts:45]: I think you should be able to make this an optional parameter.

```suggestion
  identityKey?: string,
```

File: test/integration/e2e/admin-api/postUsers.test.ts [hunk 2]
```diff
@@ -48,6 +48,37 @@ describe('E2E - adminApiRouter - POST /users', function (): void {
       .expect(HttpStatus.Created, done)
   })
 
+  it('Returns a 201 when creating a new user with an identity key', function (done): void {
+    const payId = 'jacksmith$xpring.money'
+    const identityKey = `Imp3ayI6IHsKICAia3R5IjogIkVDIiwgCiAgInVzZSI6ICJzaWciL`
+
+    // GIVEN a user with a PayID known to not exist on the PayID service
+    const userInformation = {
+      payId: payId.toUpperCase(),
```
Comment (hbergren) [inline test/integration/e2e/admin-api/postUsers.test.ts:57]: probably don't need to `toUpperCase()` this.

Comment (austin-king) [reply test/integration/e2e/admin-api/postUsers.test.ts]: Yup, you're right, removed

File: src/data-access/users.ts (hunk 1 above)
Comment (austin-king) [reply src/data-access/users.ts]: Lol. Yes. You are right, fixed this.

**Comment by austin-king** [comment]
> I would add another test, for a Admin API GET (if that is implemented yet).
> 
> We could add a seeded value for a PayID with an identity key, and make sure that we can retrieve it.

Good point, will do that now.

**Comment by austin-king** [comment]
> > I would add another test, for a Admin API GET (if that is implemented yet).
> > We could add a seeded value for a PayID with an identity key, and make sure that we can retrieve it.
> 
> Good point, will do that now.

Oh actually, we were planning to implement that in another PR through this task: https://app.asana.com/0/1182225689526777/1186929984905890

I can take that on next if we want to make sure any issues that might be caused by this PR are addresses as soon as possible.

**Pull request merged**